### ISSUE TITLE
Possible TSI index leakage from other database

### ISSUE THREAD

**Issue by blind-oracle** [influxdata/influxdb#9089]
### Bug report

__System info:__
InfluxDB 1.3.7 with tsi1
CentOS 6

__Steps to reproduce:__
We have custom collectors that receive nginx access syslog in a specific format and then pushes this data into InfluxDB through telegraf.
Then we use a continuous query to aggregate this data for long storage in a seprarate table with different retention policy.

We're successfully ingesting the data in the initial table, it's indexed correctly:
```
> show tag values on nginx from nginx_frontend with key = "Extension"
name: nginx_frontend
key       value
---       -----
Extension Unknown
Extension css
Extension eot
Extension html
Extension ico
Extension jpeg
Extension jpg
Extension js
Extension manifest
Extension mp4
Extension png
Extension svg
Extension ttf
Extension txt
Extension woff
Extension woff2
```

Then the following CQ kicks in:
```
CREATE CONTINUOUS QUERY nginx_frontend_aggregated ON nginx
BEGIN
SELECT count(ClientIP), sum(*), mean(*)
INTO nginx.long.nginx_frontend_aggregated
FROM nginx.short.nginx_frontend GROUP BY time(1m), *
END
```

And in the resulting table we get nightmare:
```
> show tag values on nginx from nginx_frontend_aggregated with key = "Extension"
name: nginx_frontend_aggregated
key       value
---       -----
Extension 1726783
Extension 1917019
Extension 79031262653
Extension 79037750518
Extension 79154529305
Extension 79163304063
Extension 79164076807
Extension 79169956077
Extension 79057741721
Extension 79175923051
Extension 79266863148
Extension 79168344685
Extension 1892230
...
```
Here we have a *ton* of some weird Extensions that are not in the original table.
We have a separate database that deals with phone numbers and I suspect that tag values are somehow leaked from there. However we don't have any overlapping CQs that can push the data from DB to another DB - I've checked twice.

And what's most interesting - the data with the corresponding tag values isn't in the table:

Real tag value:
```
> select sum(count) from nginx.long.nginx_frontend_aggregated where Extension = 'js'
name: nginx_frontend_aggregated
time sum
---- ---
0    2836
```

Leaked tag value:
```
> select sum(count) from nginx.long.nginx_frontend_aggregated where Extension = '79269889379'
> 
```
This behavior is 100% reproducible on my setup.
I'm dropping both measurements, pour the data again and here it is.

Hint: we have different table in this database which differs only in a few fields/tags and it's CQ only differs by tables names - this table does not exhibit the mentioned behavior.

__Expected behavior:__ Have tag values according to actual data

__Actual behavior:__ Have weird tag values from separate DB

**Comment by e-dard** [comment]
Hi @blind-oracle, thanks for the report.

When you say it's 100% reproducible, do you mean you can reproduce it from a fresh db given some set of input data? Would it be possible to share those reproduction steps? (You can also share things privately with us if there are sensitive data in there).

Another thing to note here: `SHOW TAG VALUES` will return the values from _all_ retention policies. Is it possible you have another retention policy on the `nginx` database that could contain those values for the `Extension` tag?

It's also possible that possibly you did in the past, but you don't any longer, and maybe there is a bug that prevented them from being removed from the TSI index properly.

A better understanding of how we can reproduce this issue would help us track it down quickly.

**Comment by blind-oracle** [comment]
> When you say it's 100% reproducible, do you mean you can reproduce it from a fresh db given some set of input data? Would it be possible to share those reproduction steps? (You can also share things privately with us if there are sensitive data in there).
Nope, it was only reproducible on this specific database.

In the continuation of these problems we had a strange disk space outage on the InfluxDB host that was caused by a *single shard* of this database - it started to grow with a speed of about 1Gb/minute until it consumed all available space. Our rate of incoming data is not nearly as big as this growth.

After that, InfluxDB crashed and after restart it was:
a) complaining about a checksum corruption in one database
b) failing to do compaction in this problematic database
c) failing to remove the database

So I had to remove these 2 databases by hand after stopping InfluxDB and recreate them from scratch. After that the bug disappeared and now I'm getting correct tag values.

> Another thing to note here: SHOW TAG VALUES will return the values from all retention policies. Is it possible you have another retention policy on the nginx database that could contain those values for the Extension tag?

Nope, these fake tag values consistently appeared *only* after the CQ executed for the first time.
So:
1. Created CQ
2. Ingested some data to the original measurement with "short" retention period
3. Looked at SHOW TAG VALUES - it' ok
4. Waited for a 1 minute for CQ to run and aggregate data into a table with "long" retention period
5. Looked at SHOW TAG VALUES again - damn, it's flooded with fake values
6. Drop the aggregated table and looked at SHOW TAG VALUES - the fake ones are gone
So it's clear to me that this particular measurement contained the problematic tag values.

> It's also possible that possibly you did in the past, but you don't any longer, and maybe there is a bug that prevented them from being removed from the TSI index properly.
It's possible, but I haven't touched this database for quite a long time after upgrading to the tsi1 indexes. So it's hard to say what caused it.
